src/plugins/vis_types/xy/public/config/get_config.ts

```diff
@@ -91,15 +95,9 @@ export function getConfig(table: Datatable, params: VisParams): VisConfig {
  */
 const shouldEnableHistogramMode = (
   seriesParams: SeriesParam[],
-  yAspects: Aspect[],
   yAxes: Array<AxisConfig<ScaleContinuousType>>
 ): boolean => {
-  const bars = seriesParams.filter(({ type, data: { id: paramId } }) => {
-    return (
-      type === ChartType.Histogram && yAspects.find(({ aggId }) => aggId === paramId) !== undefined
-    );
-  });
-
+  const bars = seriesParams.filter(({ type }) => type === ChartType.Histogram);
```

Comment:
~To have the same histogram enablement this needs to filter out `seriesParams` that are hidden, Otherwise this logic is pointless. I don't see a way to link a `seriesParam` to the corresponding `aspect` which is what tells you it is hidden. Do you know how we can determine what `seriesParams` are hidden in the **Data** tab?~

Nevermind, the histogram mode is correct when you check the `show` property, then early return for single visible bar.

```suggestion
  const bars = seriesParams.filter(({ type, show }) => show && type === ChartType.Histogram);
  if (bars.length === 1) return true;
```

https://user-images.githubusercontent.com/19007109/132555825-5c78c1ff-32f2-40b0-94bd-f578c092a3af.mp4

Reply:
Changed code, based on your nit, but with some modifications. Thanks for highlighting this case.

Reply:
Just tested latest changes from 4c283a2356da5ed19927f8f2a2826b3188ee5354 and works perfectly!